awx/main/tests/functional/conftest.py

```diff
@@ -351,9 +352,27 @@ def kube_credential(credentialtype_kube):
     )
 
 
+# @pytest.fixture
+# def inventory(organization):
+#     return organization.inventories.create(name="test-inv")
+
+
 @pytest.fixture
 def inventory(organization):
-    return organization.inventories.create(name="test-inv")
+    available_providers = compute_cloud_inventory_sources()
+
+    if available_providers:
```

Comment:
The tests shouldn't have any complex logic, the fewer checks the better. The checks for prerequisite expectations can be assertions:
```suggestion
    assert len(available_providers), 'cloud inventory plugins must be installed'
```
But that's about it. Otherwise, you'll end up with the need to test logic in your tests, which is not something common.

Fixtures are mostly used as an Arrange part of AAA: https://jamescooke.info/arrange-act-assert-pattern-for-python-developers.html / https://jamescooke.info/aaa-part-2-extracting-arrange-code-to-make-fixtures.html / https://docs.pytest.org/en/stable/explanation/anatomy.html.

Then, the tests can rely on the fixture to set everything up correctly. If some expectation isn't met, it'll still fail in the test run, we don't need to add a lot of nested checks for those things.